homeassistant/components/dwd_weather_warnings/sensor.py

```diff
@@ -65,190 +63,110 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     api = DwdWeatherWarningsAPI(region_name)
 
-    sensors = [
-        DwdWeatherWarningsSensor(api, name, condition)
-        for condition in config[CONF_MONITORED_CONDITIONS]
-    ]
+    # Build sensor list and activate update only for the first one
+    sensors = []
+    call_update = True
+    for sensor_type in config[CONF_MONITORED_CONDITIONS]:
+        sensors.append(DwdWeatherWarningsSensor(api, name, sensor_type, call_update))
+        call_update = False
 
     add_entities(sensors, True)
 
 
 class DwdWeatherWarningsSensor(Entity):
     """Representation of a DWD-Weather-Warnings sensor."""
 
-    def __init__(self, api, name, variable):
+    def __init__(self, api, name, sensor_type, call_update):
         """Initialize a DWD-Weather-Warnings sensor."""
         self._api = api
         self._name = name
-        self._var_id = variable
-
-        variable_info = MONITORED_CONDITIONS[variable]
-        self._var_name = variable_info[0]
-        self._var_units = variable_info[1]
-        self._var_icon = variable_info[2]
+        self._sensor_type = sensor_type
+        self._call_update = call_update
 
     @property
     def name(self):
         """Return the name of the sensor."""
-        return f"{self._name} {self._var_name}"
+        return f"{self._name} {MONITORED_CONDITIONS[self._sensor_type][0]}"
 
     @property
     def icon(self):
         """Icon to use in the frontend, if any."""
-        return self._var_icon
+        return MONITORED_CONDITIONS[self._sensor_type][2]
 
     @property
     def unit_of_measurement(self):
         """Return the unit the value is expressed in."""
-        return self._var_units
+        return MONITORED_CONDITIONS[self._sensor_type][1]
 
     @property
     def state(self):
         """Return the state of the device."""
-        try:
-            return round(self._api.data[self._var_id], 2)
-        except TypeError:
-            return self._api.data[self._var_id]
+        if self._sensor_type == "current_warning_level":
+            warning_level = self._api.current_warning_level
+        else:
+            warning_level = self._api.expected_warning_level
+        return warning_level
```

Comment:
This can be streamlined:

```python
if self._sensor_type == "current_warning_level":
    return self._api.current_warning_level
return self._api.expected_warning_level
```

Reply:
I'' do that.